mapeditor/resourceExtractor/ResourceConverter.h

```diff
@@ -24,26 +24,35 @@ struct ExtractionOptions
 {
 	bool extractArchives = false;	// if set, original H3 archives will be extracted into a separate folder
 	ConversionOptions conversionOptions;
+	bool moveExtractedArchivesToSoDMod = false;
 };
 
+/// <summary>
+/// Class functionality to be used after extracting original H3 resources and before moving those resources to SoD Mod
+/// Splits def files containing individual images, so that faction resources are independent. (TwCrPort, CPRSMALL, FlagPort, ITPA, ITPt, Un32 and Un44)
+/// (town creature portraits, hero army creature portraits, adventure map dwellings, small town icons, big town icons,
+/// hero speciality small icons, hero speciality large icons)
+/// Converts all PCX images to PNG
+/// </summary>
 class ResourceConverter
 {
 
 public:
 
-	// Splits def files that are shared between factions and converts pcx to bmp depending on Extraction Options
+	// Splits def files that are shared between factions and converts pcx to PNG depending on Extraction Options
 	static void convertExtractedResourceFiles(ConversionOptions conversionOptions);
 
 private:
 
-	// converts all pcx files from /Images into PNG
-	static void doConvertPcxToPng(bool deleteOriginals);
+	// Converts all .pcx from extractedFolder/Images into .png
```

Comment:
You could have used `///` here since this is documentation for a method. Nothing critical though.

Reply:
You mean with summary and all ... or just ///?

Reply:
```suggestion
	/// Converts all .pcx from extractedFolder/Images into .png
```

Generally, preferred syntax for documenting stuff is this:
```
    /// This function does cool stuff
    void makeFoo()
```
- comment that starts with triple-slash (`///`). This comment should describe *what* function does (and not *why* or *how*).
- If necessary, add description for meaning of function parameters. 
- If comment is too long for a single line, split it into multiple line but keep triple-slash on all lines
- immediately after comment, with no empty lines - definition of described function

Reply:
shouldn't good code be self-documenting tho?

Reply:
With extremely simple code - yes.
There is largely no need to document for example Point class
```
/// Construct point from two coordinates. Surprised?
Point( int x, int y)
```
But look here:
https://github.com/vcmi/vcmi/blob/develop/lib/spells/effects/Effect.h
It is well-written code, but without good description of *all* involved classes it is hard to understand what exactly the code does. Keep in mind, all comments there were added by me recently, not by original author.


For comments in code (e.g. inside methods) general rule is that code should be self-documenting, meaning, that *what* code does should be obvious without additional comments, but describing *why* code does it this way is still usually a good idea.

Reply:
Also, @krs0 slight clarification:
Just `///`. We do use doxygen-style comments for documentation, but we rarely use doxygen-style *syntax*.

Personally don't see much point behind it especially since we don't use doxygen. 
However doxygen-style comments usually have different syntax highlight rules in IDE's, making them easily distinguishable from non-documentation comments.